src/python_testing/TC_IDM_3_2.py

```diff
@@ -0,0 +1,397 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+
+from mobly import asserts
+
+import matter.clusters as Clusters
+from matter.clusters import ClusterObjects as ClusterObjects
+# from matter.exceptions import ChipStackError
+from matter.interaction_model import InteractionModelError, Status
+from matter.testing import global_attribute_ids
+from matter.testing.basic_composition import BasicCompositionTests
+from matter.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+
+
+class TC_IDM_3_2(MatterBaseTest, BasicCompositionTests):
+    """Test case for IDM-3.2: Write Response Action from DUT to TH. [{DUT_Server}]"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.endpoint = 0
+
+    # Instead of hardcoding UnitTesting.TimedWriteBoolean, we will try to find real timed-write attributes
+    async def find_timed_write_attribute(self, endpoints_data):
+        """Find an attribute that requires timed write on the actual device"""
+        logging.info(f"Searching for timed write attributes across {len(endpoints_data)} endpoints")
+
+        for endpoint_id, endpoint in endpoints_data.items():
+            for cluster_type, cluster_data in endpoint.items():
+                cluster_id = cluster_type.id
+
+                cluster_type_enum = global_attribute_ids.cluster_id_type(cluster_id)
+                if cluster_type_enum not in [global_attribute_ids.ClusterIdType.kStandard,
+                                             global_attribute_ids.ClusterIdType.kTest]:
+                    continue
+
+                # Check each attribute in this cluster
+                all_attrs = ClusterObjects.ALL_ATTRIBUTES.get(cluster_id, {})
+
+                for attr_id, attr_class in all_attrs.items():
+                    try:
+                        # Create an instance of the attribute to check its must_use_timed_write property
```

Comment:
It's a class property, so I don't think you need to jump through all these hoops. You should just be able to call it. It defaults to false in the cluster attribute descriptor so it will always be there.

https://github.com/project-chip/connectedhomeip/blob/master/src/controller/python/matter/clusters/ClusterObjects.py#L344